web/src/Root.tsx

```diff
@@ -0,0 +1,672 @@
+  };
+
+  const ping = async (tries = 0) => {
+    let apps = state.appsList;
+    const appSlugs = apps?.map((a) => a.slug);
+    const url = `${process.env.API_ENDPOINT}/ping?slugs=${appSlugs}`;
+    await fetch(
+      url,
+      {
+        headers: {
+          Authorization: Utilities.getToken(),
+          "Content-Type": "application/json",
+        },
+      },
+      10000
+    )
+      .then(async (result) => {
+        if (result.status === 401) {
+          Utilities.logoutUser();
+          return;
+        }
+        const body = await result.json();
+        setState({
+          connectionTerminated: false,
+          snapshotInProgressApps: body.snapshotInProgressApps,
+        });
+      })
+      .catch(() => {
+        if (tries < 2) {
+          setTimeout(() => {
+            ping(tries + 1);
+          }, 1000);
+          return;
+        }
+        setState({
+          connectionTerminated: true,
+          snapshotInProgressApps: [],
+        });
+      });
+  };
+
+  const onRootMounted = () => {
+    fetchKotsAppMetadata();
+    ping();
+    checkIsHelmManaged();
+
+    if (Utilities.isLoggedIn()) {
+      getAppsList().then((appsList) => {
+        if (appsList.length > 0 && window.location.pathname === "/apps") {
+          const { slug } = appsList[0];
+          history.replace(`/app/${slug}`);
+        }
+      });
+    }
+  };
+
+  useEffect(() => {
+    onRootMounted();
+    interval = setInterval(ping, 10000);
+  }, []);
+
+  useEffect(() => {
+    if (state.connectionTerminated) {
+      clearInterval(interval);
+    } else {
+      interval = setInterval(ping, 10000);
+    }
+
+    return () => {
+      clearInterval(interval);
+    };
+  }, [state.connectionTerminated]);
+
+  const isGitOpsSupported = () => {
+    const apps = state.appsList;
+    return !!find(apps, (app) => app.isGitOpsSupported);
+  };
+
+  const isIdentityServiceSupported = () => {
+    const apps = state.appsList;
+    return !!find(apps, (app) => app.isIdentityServiceSupported);
+  };
+
+  const isGeoaxisSupported = () => {
+    const apps = state.appsList;
+    return !!find(apps, (app) => app.isGeoaxisSupported);
+  };
+
+  const isSnapshotsSupported = () => {
+    const apps = state.appsList;
+    return !!find(apps, (app) => app.allowSnapshots);
+  };
+
+  const onLogoutError = (message: string) => {
+    setState({
+      errLoggingOut: message,
+    });
+  };
+
+  return (
+    <QueryClientProvider client={queryClient}>
+      <Helmet>
+        <meta
+          httpEquiv="Cache-Control"
+          content="no-cache, no-store, must-revalidate"
+        />
+        <meta httpEquiv="Pragma" content="no-cache" />
+        <meta httpEquiv="Expires" content="0" />
+        {state.appLogo && (
+          <link rel="icon" type="image/png" href={state.appLogo} />
+        )}
+        {state.appBranding?.fontFaces &&
+          state.appBranding.fontFaces.map((fontFace, index) => (
+            <style
+              type="text/css"
```

Comment:
I removed the `rel` attribute. I was getting a type error and as far as i can tell it's not something that's available on the `<style>` tag. please double check that.

Reply:
I think rel is not necessary on the style tag